pkg/networkservice/common/monitor/client_filter.go

```
@@ -37,6 +37,9 @@ func newClientFilter(client networkservice.MonitorConnection_MonitorConnectionsC
 
 func (c *clientFilter) Recv() (*networkservice.ConnectionEvent, error) {
 	for {
+		if c == nil || c.MonitorConnection_MonitorConnectionsClient == nil {
+			return nil, nil
```

Should we return an error here to prevent a nil pointer on the receiving side?